ui/raidboss/data/06-ew/trial/zodiark-ex.ts

```diff
@@ -19,6 +192,401 @@ const triggerSet: TriggerSet<Data> = {
+            data.activePythons.splice(i, 1);
+        }
+      },
+    },
+    {
+      id: 'ZodiarkEx Blue Tether',
+      type: 'Tether',
+      netRegex: NetRegexes.tether({ id: '00A4', source: 'Zodiark' }),
+      promise: async (data, matches) => {
+        const portalActors = await fetchCombatantsByTargetID([matches.targetId]);
+        for (const actor of portalActors) {
+          if (actor.ID) {
+            // console.log('blue ', { x: actor.PosX, y: actor.PosY, typeId: '67E6', npcId: actor.ID.toString(16) });
+            data.activeSigils.push({ x: actor.PosX, y: actor.PosY, typeId: '67E6', npcId: actor.ID.toString(16) });
+          }
+        }
+      },
+      alertText: (data, matches, output) => {
+        const target = data.activeSigils[data.activeSigils.length - 1];
+        if (!target) {
+          console.error(`Blue Tether: Missing target, ${JSON.stringify(matches)}`);
+          return;
+        }
+
+        if (target.x < 100)
+          return output.west!();
+
+        if (target.x > 100)
+          return output.east!();
+
+        if (target.y < 100)
+          return output.north!();
+        return output.south!();
+      },
+      outputStrings: directionOutputStrings,
+    },
+    {
+      id: 'ZodiarkEx Red Tether',
+      type: 'Tether',
+      netRegex: NetRegexes.tether({ id: '00AB', source: 'Zodiark' }),
+      promise: async (data, matches) => {
+        const portalActors = await fetchCombatantsByTargetID([matches.targetId]);
+        for (const actor of portalActors) {
+          if (actor.ID) {
+            // console.log('red ', { x: actor.PosX, y: actor.PosY, typeId: '67E5', npcId: actor.ID.toString(16) });
+            data.activeSigils.push({ x: actor.PosX, y: actor.PosY, typeId: '67E5', npcId: actor.ID.toString(16) });
+          }
+        }
+      },
+      alertText: (data, matches, output) => {
+        const target = data.activeSigils[data.activeSigils.length - 1];
+        if (!target) {
+          console.error(`Red Tether: Missing target, ${JSON.stringify(matches)}`);
+          return;
+        }
+
+        if (target.x < 100)
+          return output.east!();
+
+        if (target.x > 100)
+          return output.west!();
+
+        if (target.y < 100)
+          return output.south!();
+        return output.north!();
+      },
+      outputStrings: directionOutputStrings,
+    },
+    {
+      id: 'ZodiarkEx Roiling Darkness Spawn',
+      type: 'AddedCombatant',
+      netRegex: NetRegexes.addedCombatant({ name: 'Roiling Darkness', capture: false }),
+      suppressSeconds: 1,
+      response: Responses.killAdds(),
+    },
+    {
+      // 67E4 Green Beam, 67E5 Rectangle, 67E6 Wedge
+      id: 'ZodiarkEx Arcane Sigil Start',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67E4', '67E5', '67E6'], source: 'Arcane Sigil' }),
+      run: (data, matches, _output) => {
+        if (parseFloat(matches.y) < 100)
```

Comment:
This is only for the triple front portals these need tracking to determine the strategy for the add phase and in/out, out/in outside of add phase.

I didn't think this through for big portals in front because I kinda hoped these were also tethered with a tiny little tether but they aren't.

If they are not part of a 3stack they need immediate call out at best, at worst they are a special case for ZodiarkEx Arcane Sigil Start Cleanup .2 seconds later when I'm sure they are not just part of a triple stack.

Or would it be better to just track add phase depending on add spawn and 3x add death? I don't know.

Reply:
Yeah, I think you could just track add phase with Roiling Darkness spawn and Apomnemoneumata (6D7A) to finish it.

Reply:
On the other hand this handles the 3 beams neatly throughout the fight so I'm more in favor of a special case for single front portal at the moment.

Except you insist.

Reply:
No, that's fine if you want to do it that way too.